### ISSUE TITLE
Negative gold when buying from store

### ISSUE THREAD

**Issue by angband-import** [angband/angband#974]
**Reported by takkaria on 30 Jan 2008 01:10 UTC**
http://angband.oook.cz/forum/showthread.php?p=5151

**Comment by angband-import** [comment]
**Comment by roustk\@alumni.caltech.edu on 2 Feb 2008 14:27 UTC**
See thread for full discussion.

The bug involves buying wands/staffs and the fact that the player always 
buys a wad with above-average charges (and thus above-average price).

In store.c (revision 681), line 1908 checks that the player can afford the 
average price in the stack, but line 1985 deducts the true price without 
checking that it is affordable.

Suggested fix: at line 1985 add the following
```
                /* Spend the money */
                p_ptr->au -= price;

                /* Hack -- force au to be non-negative */ 
                /* Player may not be able to afford above-average wand, 
                   even if they can afford an average one. */
                if (p_ptr->au < 0) p_ptr->au = 0L;
```

Kevin

**Comment by angband-import** [comment]
**Modified by cunninggabe on 18 Mar 2008 22:37 UTC**

**Comment by angband-import** [comment]
**Comment by cunninggabe on 18 Mar 2008 22:37 UTC**
Fixed in [b1b42b4] (SVN r793).
